src/main/java/net/neoforged/neoforge/network/handling/ClientPayloadContext.java

```diff
@@ -67,34 +47,14 @@ public void finishCurrentTask(Type type) {
         throw new UnsupportedOperationException("Attempted to complete a configuration task on the client!");
     }
 
-    @Override
-    public ProtocolInfo<?> protocolInfo() {
-        return listener.getConnection().getInboundProtocol();
-    }
-
     @Override
     public PacketFlow flow() {
         return PacketFlow.CLIENTBOUND;
     }
 
-    @Override
-    public ConnectionProtocol protocol() {
-        return listener.protocol();
-    }
-
     @Override
     @SuppressWarnings("resource")
     public Player player() {
-        return listener.getMinecraft().player;
-    }
-
-    @Override
-    public ChannelHandlerContext channelHandlerContext() {
-        return listener.getConnection().channel().pipeline().lastContext();
-    }
-
-    @Override
-    public Connection connection() {
-        return listener.getConnection();
+        return Minecraft.getInstance().player;
```

Comment:
Shouldn't this be wrapped in an `Objects.requireNotNull` call to guarantee a non-null return?

Reply:
Maybe? It wasn't previously, and it can't be null here in general since we got the packet listener from the player

Reply:
Isn't that also usable in the config phase?

Reply:
The client player is available during the config phase

Reply:
That's not true - it's only available after `ClientboundLoginPacket` is received. Despite the name, that is a play packet.